### ISSUE TITLE
Colors for tasks that are `overdue` and `due.today` ignore `rule.precedence.color` and do not show up correctly

### ISSUE THREAD

**Issue by kdheepak** [GothenburgBitFactory/taskwarrior#2334]
#### Bug report

* What command(s) did you run?

```
task
```

* What did you expect to happen?

Colors to show up correctly based on `rule.precedence.color`:

![Screen Shot 2020-10-21 at 2 24 29 AM](https://user-images.githubusercontent.com/1813121/96693508-9006ca00-1344-11eb-90a5-2ee184f9a74a.png)

`overdue` is before `due.today`.

I've ran 4 commands here:

```bash
$ task show color.overdue
$ task color ${result of color.overdue}
$ task show color.due.today
$ task color ${result of color.due.today}
```

![Screen Shot 2020-10-21 at 2 10 22 AM](https://user-images.githubusercontent.com/1813121/96691837-9d22b980-1342-11eb-909f-3e8daa67d1bc.png)

You can see the difference between `color.overdue` and `color.due.today`. This works as expected.

Next, I run `task next` and `task 1`.

![Screen Shot 2020-10-21 at 2 14 07 AM](https://user-images.githubusercontent.com/1813121/96692300-1b7f5b80-1343-11eb-8d76-3e532a35a3b7.png)

Next, I run `task mod 1 due:today`, `task next` and `task 1`.

![Screen Shot 2020-10-21 at 2 16 21 AM](https://user-images.githubusercontent.com/1813121/96692688-921c5900-1343-11eb-8bac-7e5b87d98164.png)

* What actually happened?

Color for task shows only what is in `color.due.today` and `color.overdue` seems to be ignored.

* Paste the output of the ```task diag``` command.

```
$ task diagnostics
TASKRC override: /Users/USER/gitrepos/taskwarrior-tui/tests/data/.taskrc
TASKDATA override: /Users/USER/gitrepos/taskwarrior-tui/tests/data/.task

task 2.5.1
   Platform: Darwin

Compiler
    Version: 4.2.1 Compatible Apple LLVM 11.0.3 (clang-1103.0.32.59)
       Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
 Compliance: C++11

Build Features
      Built: Jul  9 2020 07:32:08
      CMake: 3.17.3
    libuuid: libuuid + uuid_unparse_lower
  libgnutls: 3.6.14
 Build type: Release

Configuration
       File: /Users/USER/gitrepos/taskwarrior-tui/tests/data/.taskrc (found), 1831 bytes, mode 100644
       Data: /Users/USER/gitrepos/taskwarrior-tui/tests/data/.task (found), dir, mode 40755
     TASKRC: /Users/USER/gitrepos/taskwarrior-tui/tests/data/.taskrc
   TASKDATA: /Users/USER/gitrepos/taskwarrior-tui/tests/data/.task
    Locking: Enabled
         GC: Enabled
    $VISUAL: nvim
     Server:
      Trust: strict
Certificate: , not readable, 0 bytes
        Key: , not readable, 0 bytes
    Ciphers: NORMAL
      Creds:

Hooks
     System: Enabled
   Location: /Users/USER/gitrepos/taskwarrior-tui/tests/data/.task/hooks
             (-none-)

Tests
      $TERM: xterm-256color (120x30)
       Dups: Scanned 14 tasks for duplicate UUIDs:
             No duplicates found
 Broken ref: Scanned 14 tasks for broken references:
             No broken references found

```

I've attached my `.task` folder and `.taskrc` file in this repository:

https://github.com/kdheepak/taskwarrior-testdata/commit/d0d123f989bb67193e0b60a7396a00bc816d5910

I get the same behavior with the latest release as well as the `master` branch of this repository.

**Comment by kdheepak** [comment]
I did some print line style debugging and found that the precedence is in fact correctly being calculated, and is what I expect it to be at this point in the code.

https://github.com/GothenburgBitFactory/taskwarrior/blob/01696a307b6785be973e3e6428e6ade2a3872c1e/src/rules.cpp#L76

I wanted to check with the developers if they had an idea of what might be going on before jumping in with a debugger to figure this out.

**Comment by tbabej** [comment]
@kdheepak No immediate idea wat might be going on. Further investigation from your end would be very welcome.

**Comment by kdheepak** [comment]
Thanks for the comment. If I look into this further I will post here!

**Comment by kdheepak** [comment]
@tbabej I think I figured out why this is happening:

Here's the check for adding the `OVERDUE` tag:

https://github.com/GothenburgBitFactory/taskwarrior/blob/2f47226f91f0b02f7617912175274d9eed85924f/src/Task.cpp#L541-L559

Here's how the color is applied when checking if the task is overdue is `OVERDUE`:

https://github.com/GothenburgBitFactory/taskwarrior/blob/2f47226f91f0b02f7617912175274d9eed85924f/src/rules.cpp#L250-L260

You can see that in `colorizeOverdue`, the check is only if it occurs in a "date before today". I think it should be modified to be the same as the `OVERDUE` tag.

Thoughts? Is there a reason this was done the way it is done?

If it is a bug, I can submit a PR. If it is intentional, can you explain the rationale? You can also close this issue in that case.

**Comment by tbabej** [comment]
@kdheepak I think your analysis is correct here. I seem to remember coming across this (and other) colorize rules the other day, and I vaguely recall that `+OVERDUE` <-> `colorizeOverdue` might not be the only place where the definition of the virtual tag does not correspond to the color rule.

EDIT: Looks like I might have already fixed this two months ago: 2adc5994d54bff4bd8c9963f477f7f4242a9287c, but it would not hurt to double check / check other color rules for correct definitions with respect to the corresponding virtual tags.

**Comment by kdheepak** [comment]
It looks like in the most recent commit you did fix it to use the method on `task` i.e. `task.is_overdue()`:

https://github.com/GothenburgBitFactory/taskwarrior/blob/cdbab698e330fe0f896d3f12b27dd7286748fec1/src/rules.cpp#L238-L244

This should resolve this issue when 2.6.0 is released! I'll close this. 

Thank you for all your work on taskwarrior!
